### PR TITLE
feat: reduce object sizes up to 87%

### PR DESCRIPTION
**Pull request by laresbernardo**
Based on demo data, 1 trial and 1000 iterations, reduced `OutputModels` from 40Mb to 5.3Mb (87% size reduction) by not storing the `xDecompVec` data.frame which contained all transformed values for all models, including carryover and immediate effects. Now we re-calculate when exporting the objects (`OutputCollect`) with little to no performance tradeoff, and reducing it's size substantially too.

### CODE REVIEW

File: R/R/pareto.R [hunk 1]
```diff
@@ -473,7 +473,29 @@ robyn_pareto <- function(InputCollect, OutputModels,
...
       ## 7. Immediate vs carryover response
-      temp <- filter(xDecompVecImmCarr, .data$solID == sid)
+      # temp <- filter(xDecompVecImmCarr, .data$solID == sid)
+      hypParamSam <- resultHypParam[resultHypParam$solID == sid, ]
+      dt_saturated_dfs <- run_transformations(InputCollect, hypParamSam, adstock)
+      coefs <- xDecompAgg$coef[xDecompAgg$solID == sid]
+      names(coefs) <- xDecompAgg$rn[xDecompAgg$solID == sid]
+      decompCollect <- model_decomp(
+        coefs = coefs,
+        y_pred = dt_saturated_dfs$dt_modSaturated$dep_var, # IS THIS RIGHT?
```
Comment (laresbernardo) [inline R/R/pareto.R:483]: @gufengzhou please, check this

**Comment by laresbernardo** [comment]
Ran one more test in my local computer, with 5 trials and 2000 iterations, and the size for `OutputModels` was:
- main: 391.9Mb (14.5 minutes)
- PR: 48.8Mb (13.7 minutes)

## IMPROVEMENT: 87.55% lighter!!

File: R/R/pareto.R (hunk 1 above)
Comment (gufengzhou) [reply R/R/pareto.R:483]: LGTM

**Review by gufengzhou — APPROVED** [review]
LGTM

**Comment by gufengzhou** [comment]
> 

Can confirm the %size reduction of OutputModels is reproducible. Good job! 
For OutputCollect it'd probably need further reduction because its size is more related to the stored plotting objects.

**Pull request merged**